extensions/native/recursion/src/fri/mod.rs

```diff
@@ -193,55 +194,60 @@ pub fn verify_batch<C: Config>(
     // For each sibling in the proof, reconstruct the root.
     let left: Ptr<C::N> = builder.uninit();
     let right: Ptr<C::N> = builder.uninit();
-    builder.range(0, proof.len()).for_each(|i, builder| {
-        let sibling = builder.get_ptr(&proof, i);
-        let bit = builder.get(&index_bits, i);
-
-        builder.if_eq(bit, C::N::ONE).then_or_else(
-            |builder| {
-                builder.assign(&left, sibling);
-                builder.assign(&right, root_ptr);
-            },
-            |builder| {
-                builder.assign(&left, root_ptr);
-                builder.assign(&right, sibling);
-            },
-        );
+    builder
+        .zip(&[
+            Box::new(proof.clone()) as Box<dyn ArrayLike<C>>,
+            Box::new(index_bits.clone()) as Box<dyn ArrayLike<C>>,
+        ])
```

Comment:
I wonder if it's worth making a macro so that you can do
```
compile_zip!(builder, proof, index_bits).for_each(|ptr_vec, builder| { ... });
```
The `Box` stuff is pretty ugly.